Orange/widgets/visualize/owlineplot.py

```diff
@@ -0,0 +1,762 @@
+        self.graph.view_box.remove_profiles()
+
+    def _plot_group(self, data, indices, index=None):
+        color = self.__get_group_color(index)
+        group = ProfileGroup(data, indices, color, self.graph)
+        kwargs = self.__get_visibility_flags()
+        group.set_visible_error(**kwargs)
+        group.set_visible_mean(**kwargs)
+        group.set_visible_range(**kwargs)
+        group.set_visible_profiles(**kwargs)
+        self.__groups.append(group)
+
+    def __get_group_color(self, index):
+        if self.group_var is not None:
+            return QColor(*self.group_var.colors[index])
+        return QColor(LinePlotStyle.DEFAULT_COLOR)
+
+    def __get_visibility_flags(self):
+        return {"show_profiles": self.show_profiles,
+                "show_range": self.show_range,
+                "show_mean": self.show_mean,
+                "show_error": self.show_error}
+
+    def _update_profiles_color(self):
+        # color alpha depends on subset and selection; with selection or
+        # subset profiles color has more opacity
+        if not self.show_profiles:
+            return
+        for group in self.__groups:
+            has_sel = bool(self.subset_indices) or bool(self.selection)
+            group.update_profiles_color(has_sel)
+
+    def _update_sel_profiles_and_range(self):
+        # mark selected instances and selected range
+        if not (self.show_profiles or self.show_range):
+            return
+        for group in self.__groups:
+            inds = [i for i in group.indices if self.__in(i, self.selection)]
+            table = self.data[inds, self.graph_variables].X if inds else None
+            if self.show_profiles:
+                group.update_sel_profiles(table)
+            if self.show_range:
+                group.update_sel_range(table)
+
+    def _update_sel_profiles_color(self):
+        # color depends on subset; when subset is present,
+        # selected profiles are black
+        if not self.selection or not self.show_profiles:
+            return
+        for group in self.__groups:
+            group.update_sel_profiles_color(bool(self.subset_indices))
+
+    def _update_sub_profiles(self):
+        # mark subset instances
+        if not (self.show_profiles or self.show_range):
+            return
+        for group in self.__groups:
+            inds = [i for i, _id in zip(group.indices, group.ids)
+                    if self.__in(_id, self.subset_indices)]
+            table = self.data[inds, self.graph_variables].X if inds else None
+            group.update_sub_profiles(table)
+
+    def _update_visibility(self, obj_name):
+        if self.__groups is None:
+            return
+        self._update_profiles_color()
+        self._update_sel_profiles_and_range()
+        self._update_sel_profiles_color()
+        for group in self.__groups:
+            kwargs = self.__get_visibility_flags()
+            getattr(group, "set_visible_{}".format(obj_name))(**kwargs)
```

Comment:
`kwargs` can be before the loop. Better still:

```
        shown = getattr(self, f"show_{obj_name}")
        for group in self.__groups:
            getattr(group, f"set_visible_{obj_name}")(shown)
```

You can then remove method `__get_visibility_flags()`. More importantly, you can then remove argument `**_` from `ProfileGroup`'s methods.